### ISSUE TITLE
amd-smi reports ALL N/A on Strix Halo gfx1151 — kernel exposes data, user-space tools are blind

### ISSUE THREAD

**Issue by bkpaine1** [RadeonOpenCompute/ROCm#6035]
## Summary

`amd-smi` v26.2.1 on ROCm 7.2.0 reports **every monitoring metric as N/A** on the Radeon 8060S (gfx1151, Strix Halo). Power, temperature, clocks, utilization, fan speed, PCIe, power caps, thermal limits — all blind. The kernel driver (`amdgpu`) exposes the data correctly through sysfs/hwmon. `amd-smi` cannot read any of it.

This is AMD's flagship AI-branded consumer APU. It says "AI" in the product name. We're running sustained ML training workloads on it. We cannot monitor GPU health through official tools.

## Hardware
- **APU**: AMD Ryzen AI MAX+ 395
- **GPU**: Radeon 8060S (gfx1151) — integrated
- **Memory**: 128 GB unified (64 GB VRAM allocation)
- **System**: GMKTEC EVO X2

## Software
- **Kernel**: 6.18.1
- **ROCm**: 7.2.0
- **amd-smi**: 26.2.1
- **rocm-smi**: 4.0.0
- **PyTorch**: 2.11.0a0+rocm7.11.0a20260106 (TheROCk nightly)

## The Problem

During sustained ML training (5-10 minute pretraining loops), the GPU runs at 87-91°C and 120W. We have experienced **thermal reboots** under heavy load. We need to monitor thermals programmatically to build safety cutoffs.

### amd-smi monitor output (COMPLETELY BLIND):
```
GPU  XCP  POWER  PWR_CAP   GPU_T   MEM_T   GFX_CLK   GFX%   MEM%  MEM_CLOCK
  0    0    N/A      N/A     N/A     N/A       N/A    N/A    N/A        N/A
```

### amd-smi metric output (COMPLETELY BLIND):
```
USAGE: N/A
POWER:
    SOCKET_POWER: N/A
    GFX_VOLTAGE: N/A
    THROTTLE_STATUS: N/A
CLOCK: N/A
TEMPERATURE:
    EDGE: 91 °C        <-- this ONE field works
    HOTSPOT: N/A
    MEM: N/A
FAN:
    SPEED: N/A
    RPM: N/A
```

### amd-smi static LIMIT section (ALL N/A):
```
LIMIT:
    PPT0:
        MAX_POWER_LIMIT: N/A
        MIN_POWER_LIMIT: N/A
        SOCKET_POWER_LIMIT: N/A
    SLOWDOWN_EDGE_TEMPERATURE: N/A
    SHUTDOWN_EDGE_TEMPERATURE: N/A
```

### Meanwhile, sysfs/hwmon exposes the data correctly:
```
power1_average:  119095000 µW  (119.0W)    ← amd-smi says N/A
temp1_input:     89000 m°C     (89.0°C)    ← amd-smi says N/A (monitor)
freq1_input:     2787000000 Hz (2787MHz)   ← amd-smi says N/A
gpu_busy_percent: 100                       ← amd-smi says N/A
pp_dpm_sclk:     0: 600Mhz / 1: 2787Mhz* / 2: 2900Mhz  ← amd-smi says N/A
```

### rocm-smi gets PARTIAL data:
```
Temp: 91.0°C  Power: 120.019W  SCLK: N/A  Fan: 0%  PwrCap: N/A  GPU%: 100%
```
`rocm-smi` reads temp, power, and GPU% from sysfs directly, but SCLK and power cap return N/A (likely falling back to the amdsmi library for those fields).

## Root Cause

The `amdsmi` library that `amd-smi` uses does not have support for gfx1151 APU monitoring interfaces. The library appears to expect discrete GPU SMU interfaces and doesn't know how to read the APU's power management data, even though the kernel's `amdgpu` driver exposes it through standard hwmon/sysfs paths.

`rocm-smi` has partial coverage because it reads some fields directly from sysfs (temp, power via hwmon) but delegates others to amdsmi (clocks, power cap, fan).

## Impact

1. **Safety**: We cannot build programmatic thermal safeguards for ML training. We've had thermal reboots at 90°C+ during sustained workloads. Without `amd-smi` working, users must scrape raw sysfs paths — which most ML practitioners won't do.

2. **Ecosystem**: NVIDIA's `nvidia-smi` works perfectly on every GPU they sell. This is table stakes. Users evaluating AMD for ML see `amd-smi` return all N/A and conclude the hardware isn't supported. We've published benchmark data showing AMD Strix Halo beating RTX 4090 on training quality (val_bpb 1.219 vs 1.844) — but we can't tell users "just use amd-smi to monitor your training" because it doesn't work.

3. **TheROCk nightlies**: The nightly PyTorch wheels for gfx1151 work excellently. But they ship with the same broken monitoring tools. Native kernel support for ML training + completely blind monitoring = a dangerous combination.

## Reproduction

```bash
# On any Strix Halo / gfx1151 system with ROCm 7.2.0:
amd-smi monitor -p -t -u -m
# Expected: Power, temp, clocks, utilization
# Actual: ALL N/A

# Proof the data exists:
cat /sys/class/drm/card*/device/hwmon/hwmon*/power1_average
cat /sys/class/drm/card*/device/hwmon/hwmon*/temp1_input
cat /sys/class/drm/card*/device/hwmon/hwmon*/freq1_input
cat /sys/class/drm/card*/device/gpu_busy_percent
# All return valid data
```

Full diagnostic script and output: https://github.com/bkpaine1/amdsense

## What We're Asking For

1. **amd-smi gfx1151 APU support** — read power, temp, clocks, utilization, and power caps from hwmon/sysfs on Strix Halo APUs
2. **Power cap exposure** — `power1_cap` is not exposed in sysfs either. Users need to know their thermal envelope.
3. **Thermal limit exposure** — slowdown and shutdown temperatures should be queryable so ML frameworks can implement safety cutoffs

## Context

We've filed 5 bf16 precision bugs separately (#6034). This issue is specifically about the monitoring gap. We're running 100+ autonomous ML experiments on this hardware and publishing all data publicly. AMD has a real competitive advantage with unified memory APUs for ML — 128GB addressable VRAM vs NVIDIA's 24GB hard limit on consumer cards. But the tooling needs to match.

Related: Linux 7.1 is adding NPU power reporting via `DRM_IOCTL_AMDXDNA_GET_INFO` for Ryzen AI. The GPU monitoring side needs the same attention.

## Why This Matters Beyond One Bug

Every ML engineer who tries AMD for the first time runs `amd-smi` and sees all N/A. Most of them go back to NVIDIA and never return. That's not a lost hardware sale — that's a lost evangelist.

Enterprise GPU procurement doesn't start with an RFP. It starts with an engineer who trained a model on Radeon at home and brings that experience to a data center conversation. NVIDIA understands this — they're actively lending engineers to open-source AI projects (OpenClaw, etc.) because every developer who builds on CUDA at their desk becomes an advocate for NVIDIA in the procurement meeting.

AMD has silicon that beats the RTX 4090 on training quality at lower cost. The hardware argument is already won. But broken tooling on consumer/prosumer platforms ensures nobody gets far enough to make that argument at work. BlackBerry had 50% enterprise market share when the iPhone launched. They lost it not because IT departments switched, but because executives experienced something better personally and pushed it into corporate.

`amd-smi` returning all N/A on a Ryzen AI MAX+ system is the equivalent of shipping a phone with no email client and wondering why enterprise won't adopt. The data is right there in the kernel. Read it.

**Comment by chejh-amd** [comment]
amd-smi/amdsmi currently has no monitoring support for the gfx1151 APU (Strix Halo).
The kernel (amdgpu) exposes all metrics correctly through sysfs/hwmon, but amdsmi does not read them, so amd-smi shows power/clock/util/fan/limits as N/A. rocm-smi only works partially for the same reason.

**Comment by pedapudi** [comment]
Thanks a lot, @Michelle-HCJ 

Just for clarity, is the objective that amd-smi and rocm-smi will support gfx1151? sysfs/hwmon reporting being accurate is terrific (and I personally appreciate prioritizing accuracy in hwmon), so this is mostly a clarifying question. FWIW, documentation citing sysfs/hwmon for monitoring Strix Halo would go a long way.

**Comment by kraln** [comment]
```
# amd-smi
+------------------------------------------------------------------------------+
| AMD-SMI 26.2.1+fc0010cf6a    amdgpu version: Linuxver ROCm version: 7.2.0    |
| VBIOS version: 023.011.000.039.000001                                        |
| Platform: Linux Baremetal                                                    |
|-------------------------------------+----------------------------------------|
| BDF                        GPU-Name | Mem-Uti   Temp   UEC       Power-Usage |
| GPU  HIP-ID  OAM-ID  Partition-Mode | GFX-Uti    Fan               Mem-Usage |
|=====================================+========================================|
| 0000:c3:00.0    AMD Radeon Graphics | N/A        N/A   0                 N/A |
|   0       0     N/A             N/A | N/A        N/A              148/512 MB |
+-------------------------------------+----------------------------------------+
+------------------------------------------------------------------------------+
| Processes:                                                                   |
|  GPU        PID  Process Name          GTT_MEM  VRAM_MEM  MEM_USAGE     CU % |
|==============================================================================|
|    0       6910  python3.12             2.8 GB  898.4 KB     2.9 GB  N/A     |
+------------------------------------------------------------------------------+
```
➕ from me, really jarring experience. IMHO either the device should be unsupported (fail with an error, say 'use rocm-smi', etc.) or show proper values.

**Comment by darren-amd** [comment]
Hi @bkpaine1,

Thanks for reporting this issue! A lack of metrics is definitely something that we understand affects the developer experience and is something we are actively working on addressing. Will keep you updated on any progress.

**Comment by timreha** [comment]
Yes agreed! it is super critical.

**Comment by mantkiew** [comment]
Related to this 
```
xrt-smi validate
Validate Device           : [0000:c5:00.1]
    Platform              : RyzenAI-npu5
    Power Mode            : Performance
-------------------------------------------------------------------------------
Test 1 [0000:c5:00.1]     : gemm 
    Error(s)              : No archive provided, skipping test                  
    Test Status           : [FAILED]
-------------------------------------------------------------------------------
Test 2 [0000:c5:00.1]     : latency 
    Error(s)              : No archive found, skipping test                     
    Test Status           : [FAILED]
-------------------------------------------------------------------------------
Test 3 [0000:c5:00.1]     : throughput 
    Error(s)              : No archive found, skipping test                     
    Test Status           : [FAILED]
-------------------------------------------------------------------------------
Validation failed. Please run the command '--verbose' option for more details
```
Also, `aie-partitions` returns lots of `N/A`!
```
xrt-smi examine --report aie-partitions --verbose
Verbose: Enabling Verbosity
Verbose: SubCommand: examine

------------------------------
[0000:c5:00.1] : RyzenAI-npu5
------------------------------
AIE Partitions
  Total Memory Usage: N/A
  Partition Index   : 0
    Columns: [0, 1, 2, 3, 4, 5, 6, 7]
    HW Contexts:
      |PID                 |Ctx ID     |Submissions |Migrations  |Err  |Priority |
      |Process Name        |Status     |Completions |Suspensions |     |GOPS     |
      |Memory Usage        |Instr BO   |            |            |     |FPS      |
      |                    |           |            |            |     |Latency  |
      |====================|===========|============|============|=====|=========|
      |109237              |1          |0           |0           |0    |N/A      |
      |N/A                 |Active     |0           |0           |     |N/A      |
      |N/A                 |N/A        |            |            |     |N/A      |
      |                    |           |            |            |     |N/A      |
      |--------------------|-----------|------------|------------|-----|---------|
      |109237              |2          |148         |0           |0    |N/A      |
      |N/A                 |Active     |148         |0           |     |N/A      |
      |N/A                 |N/A        |            |            |     |N/A      |
      |                    |           |            |            |     |N/A      |
      |--------------------|-----------|------------|------------|-----|---------|
      |109237              |3          |0           |0           |0    |N/A      |
      |N/A                 |Active     |0           |0           |     |N/A      |
      |N/A                 |N/A        |            |            |     |N/A      |
      |                    |           |            |            |     |N/A      |
      |--------------------|-----------|------------|------------|-----|---------|
      |109237              |4          |0           |0           |0    |N/A      |
      |N/A                 |Active     |0           |0           |     |N/A      |
      |N/A                 |N/A        |            |            |     |N/A      |
      |                    |           |            |            |     |N/A      |
      |--------------------|-----------|------------|------------|-----|---------|
      |109237              |5          |1           |0           |0    |N/A      |
      |N/A                 |Active     |1           |0           |     |N/A      |
      |N/A                 |N/A        |            |            |     |N/A      |
      |                    |           |            |            |     |N/A      |
      |--------------------|-----------|------------|------------|-----|---------|
      |109237              |6          |1           |0           |0    |N/A      |
      |N/A                 |Active     |1           |0           |     |N/A      |
      |N/A                 |N/A        |            |            |     |N/A      |
      |                    |           |            |            |     |N/A      |
      |--------------------|-----------|------------|------------|-----|---------|
      |109237              |7          |464         |0           |0    |N/A      |
      |N/A                 |Active     |463         |0           |     |N/A      |
      |N/A                 |N/A        |            |            |     |N/A      |
      |                    |           |            |            |     |N/A      |
      |--------------------|-----------|------------|------------|-----|---------|
      |109237              |8          |370         |0           |0    |N/A      |
      |N/A                 |Active     |370         |0           |     |N/A      |
      |N/A                 |N/A        |            |            |     |N/A      |
      |                    |           |            |            |     |N/A      |
      |--------------------|-----------|------------|------------|-----|---------|
      |109237              |9          |12          |0           |0    |N/A      |
      |N/A                 |Active     |12          |0           |     |N/A      |
      |N/A                 |N/A        |            |            |     |N/A      |
      |                    |           |            |            |     |N/A      |
      |--------------------|-----------|------------|------------|-----|---------|
      |109237              |10         |62          |0           |0    |N/A      |
      |N/A                 |Active     |62          |0           |     |N/A      |
      |N/A                 |N/A        |            |            |     |N/A      |
      |                    |           |            |            |     |N/A      |
      |--------------------|-----------|------------|------------|-----|---------|
AIE Columns
  |Column  |HW Context Slot                  |
  |--------|---------------------------------|
  |0       |[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]  |
  |1       |[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]  |
  |2       |[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]  |
  |3       |[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]  |
  |4       |[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]  |
  |5       |[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]  |
  |6       |[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]  |
  |7       |[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]  |
```
Very embarassing for AMD. Apparently, this worked for the earlier `strix-point` but no longer for `strix-halo`.

**Comment by bkpaine1** [comment]
Thanks @mantkiew — this is a valuable corroboration, because it shows the problem isn't confined to the GPU stack.

Tying the two halves together:

**GPU side (this issue):** `amd-smi` reports all-N/A on gfx1151 even though the kernel/sysfs is exposing the underlying data — the user-space tool is blind, not the hardware.

**NPU side (your data):** `xrt-smi examine --report aie-partitions` shows the *same* signature on the Ryzen AI NPU (RyzenAI-npu5). The engine is demonstrably doing work — active HW contexts with real, non-zero submissions/completions (148, 464, 370, …) — yet every performance metric (GOPS, FPS, Latency, Total/per-context Memory Usage, even Process Name) reads N/A.

One thing worth separating so it doesn't muddy the report: the `xrt-smi validate` failures are a *different* problem — they're "No archive found, skipping test", i.e. the validation test binaries aren't installed. That's a packaging/setup gap, not the telemetry bug. The real corroboration is the all-N/A in `aie-partitions` while the contexts are clearly live.

So across **both** compute engines on Strix Halo we see the same pattern:

- the hardware is working and the kernel-level data exists,
- but the user-space telemetry/reporting layer returns N/A,
- and it's a **regression** — these metrics populated on the previous-gen Strix Point and stopped on Strix Halo.

These are two different user-space stacks (amdgpu/`amd-smi` for the GPU, XDNA/`xrt-smi` for the NPU), so the open question is whether this is **one shared root cause** — a common SMU/metrics interface or sysfs plumbing not wired up for the gfx1151 / npu5 generation — or **two independent regressions that happen to land on the same silicon**. Either way it's a systemic observability gap on Strix Halo: the platform computes fine but is effectively un-instrumentable from user space.

Happy to gather additional `--verbose` dumps, Strix Point vs Strix Halo kernel-sysfs comparisons, or driver/firmware versions to help triage. Flagging for visibility — this affects anyone trying to profile or monitor Strix Halo for GPU *or* NPU workloads.

**Comment by mantkiew** [comment]
Re: validation binaries. I have the following installed on CachyOS and there are no more XRT packages available:
```
❯ pacman -Ss xrt
cachyos-extra-znver4/xrt 1:2.21.75-9.1 [installed]
    Runtime for AIE and FPGA based platforms
cachyos-extra-znver4/xrt-plugin-amdxdna 1:2.21.75-2.1 [installed]
    Runtime for AIE and FPGA based platforms
```
I am using Lemonade with FastFlowLM to test the NPU.

**Comment by ghostklart** [comment]
question:
the `power1_cap` appearance in amdgpu driver for strix-halo/gfx1151 is to be expected following this issue, or should be a separate issue?
